diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/TypeAliasRule.kt

```diff
@@ -30,7 +32,7 @@ class TypeAliasRule(private val configRules: List<RulesConfig>) : Rule("type-ali
         emitWarn = emit
         isFixMode = autoCorrect
 
-        if (node.elementType == TYPE_REFERENCE) {
+        if (node.elementType == TYPE_REFERENCE && (node.treeParent.elementType == FUN || node.treeParent.elementType == PROPERTY)) {
```

Comment:
I think it's better to add exclusions and not inclusions: for example, this rule should check types of function parameters, which are not direct ancestors of FUN node. I think you can just check if this node is not from supertype list.